app/models/giveaway.rb

```diff
@@ -1,16 +1,22 @@
 class Giveaway < ApplicationRecord
     has_many :users
+    has_many :winners, :class_name => "User" 
     has_many :products
   
     def choose_winners   
-      winners = {} 
       all_applicants = users.to_a.shuffle
       
       products.each do |product|
-        product[:quantity].times { winners[all_applicants.pop] = product[:prod_id] }
+        product[:quantity].times do
+            w = all_applicants.pop
+            if !w 
+                return 
+            else
+                w.item_won = product[:prod_id]
+                winners << w
```

Comment:
That's awesome! It figures there's a rails-magic way of doing it :) I'll add a PR to fix this